File: apps/api/schemas/documents.py
import datetime
import uuid

from pydantic import BaseModel


class DocumentOut(BaseModel):
    id: uuid.UUID
    workspace_id: uuid.UUID
    filename: str
    content_hash_current: str
    status: str
    current_version_id: uuid.UUID | None
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = {"from_attributes": True}


class DocumentVersionOut(BaseModel):
    id: uuid.UUID
    document_id: uuid.UUID
    version_number: int
    content_hash: str
    page_count: int
    is_current: bool
    superseded_at: datetime.datetime | None
    created_at: datetime.datetime

    model_config = {"from_attributes": True}


class UploadAccepted(BaseModel):
    document_id: uuid.UUID
    document_version_id: uuid.UUID
    job_id: uuid.UUID
    status: str  # "queued"


class UploadUnchanged(BaseModel):
    document_id: uuid.UUID
    document_version_id: uuid.UUID
    status: str  # "unchanged"

File: apps/UI/src/api/documents.ts
import { apiFetch } from "./client";

export interface Document {
  id: string;
  workspace_id: string;
  filename: string;
  content_hash_current: string;
  status: string;
  current_version_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface UploadAccepted {
  document_id: string;
  document_version_id: string;
  job_id: string;
  status: "queued";
}

export interface UploadUnchanged {
  document_id: string;
  document_version_id: string;
  status: "unchanged";
}

export function listDocuments(
  accessToken: string | undefined,
  workspaceId: string,
): Promise<Document[]> {
  return apiFetch<Document[]>(`/workspaces/${workspaceId}/documents`, accessToken);
}

export function uploadDocument(
  accessToken: string | undefined,
  workspaceId: string,
  file: File,
): Promise<UploadAccepted | UploadUnchanged> {
  const formData = new FormData();
  formData.append("file", file);
  return apiFetch<UploadAccepted | UploadUnchanged>(
    `/workspaces/${workspaceId}/documents`,
    accessToken,
    { method: "POST", body: formData },
  );
}
